src/backend/embedding/extract_file/extract_file.py

```diff
@@ -0,0 +1,245 @@
+                        else:
+                            right_content.append({"text": block["text"], "y0": y0})
+
+                        all_content.append({"text": block["text"], "y0": y0})
+
+                    # Check if left and right content exists
+                    has_left = len(left_content) > 0
+                    has_right = len(right_content) > 0
+
+                    # Determine layout type
+                    if has_left and has_right:
+                        # Left/Right Layout: Process separately and merge
+                        left_content = sorted(left_content, key=lambda b: b["y0"])
+                        right_content = sorted(right_content, key=lambda b: b["y0"])
+                        merged_content = " ".join([b["text"] for b in left_content + right_content])
+                    else:
+                        # Full-Paragraph Layout: Sort all text blocks and merge them
+                        sorted_content = sorted(all_content, key=lambda b: b["y0"])
+                        merged_content = " ".join([b["text"] for b in sorted_content])
+
+                    # Append merged text content to page_elements
+                    page_elements.append({
+                        "type": "text",
+                        "content": merged_content,
+                        "x0": 0,
+                        "y0": 0  # No specific position for merged content
+                    })
+
+                    # Extract text and table data from the page
+                    tables = page.extract_tables()
+                    if tables:
+                        for table in tables:
+                            # Format and collect table data
+                            table_text = self.format_table_as_text(table)
+                            page_elements.append({
+                                "type": "table",
+                                "content": table_text,
+                                "x0": 0,
+                                "y0": 0
+                            })
+
+                    # If there's no extracted text, perform OCR and append image-based text
+                    if not page.extract_text():
+                        page_image = page.to_image(resolution=300).original
+                        ocr_text = pytesseract.image_to_string(page_image)
+                        page_elements.append({
+                            "type": "image",
+                            "content": ocr_text.strip(),
+                            "x0": 0,
+                            "y0": 0
+                        })
+
+                    # Sort page elements by vertical position (y0) and horizontal position (x0)
+                    page_elements.sort(key=lambda w: (w["y0"], w["x0"]))
+
+                    # Handle reversed vertical text
+                    corrected_content = []
+                    for element in page_elements:
```

Comment:
This logic is correct but sometimes it prints the reverse text

Can you make sure, you read the text and check whether the reverse text makes sense or the original text makes sense and then place it in the extracted text. In that way, however we read the text, the correct text will be placed in the extracted text

                    for element in page_elements:
                        if element["type"] == "text" and self.is_vertical_text(element):
                            corrected_content.append(self.reverse_vertical_text(element["content"]))
                        else:
                            corrected_content.append(element["content"])
                            
                            
                            Check this example
                            
                            
![image](https://github.com/user-attachments/assets/4a141ef2-5403-4525-9489-df4cfe27530f)